src/test/java/com/zendesk/maxwell/SavedSchemaSupport.java

```diff
@@ -0,0 +1,20 @@
+package com.zendesk.maxwell;
+
+import com.zendesk.maxwell.replication.BinlogPosition;
+import com.zendesk.maxwell.replication.Position;
+import com.zendesk.maxwell.schema.MysqlSavedSchema;
+import com.zendesk.maxwell.schema.Schema;
+
+final public class SavedSchemaSupport {
+	private SavedSchemaSupport() { }
+
+	public static MysqlSavedSchema getSavedSchema(MaxwellContext context, Schema schema, Position position) throws Exception {
+		if (context.getConfig().gtidMode) {
+			return new MysqlSavedSchema(context, schema, position);
+		}
+
+		return new MysqlSavedSchema(context.getServerID(), context.getCaseSensitivity(),
+				schema, new Position(new BinlogPosition(position.getBinlogPosition().getOffset() - 1L,
```

Comment:
sorry, I'm kinda confused;  why do we need the `getOffset() - 1` bits?

Reply:
Hi @osheroff , thanks for the question. This is only for `testing data`. The purpose of adding `- 1` is to make sure the seeding saved schema's position is prior to the current position. It fixed lots of failed tests in `MysqlSavedSchemaTest` and `RecoveryTest` (Previously, the saved schema is same position with current position. (lots of null pointer exception in the tests)). :)

Reply:
ok.  it's a little nasty, but ok.